Rubberduck.SourceControl/SourceControlProviderBase.cs

```diff
@@ -31,6 +31,8 @@ protected SourceControlProviderBase(VBProject project, IRepository repository, I
         public abstract IEnumerable<IBranch> Branches { get; }
         public abstract IList<ICommit> UnsyncedLocalCommits { get; }
         public abstract IList<ICommit> UnsyncedRemoteCommits { get; }
+        public bool NotifyExternalFileChanges { get; protected set; }
+        public bool NotifyVBAChanges { get; protected set; }
```

Comment:
That disables the add/remove/rename component handlers so they don't start messing with the repo during things like switching branches/pulls/etc.